packages/jellyfish-api-core/__tests__/category/account/getPendingFutureSwaps.test.ts

```diff
@@ -0,0 +1,333 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import BigNumber from 'bignumber.js'
+import { Testing } from '@defichain/jellyfish-testing'
+import { RegTestFoundationKeys } from '@defichain/jellyfish-network'
+
+describe('Account GetPendingFutureSwaps', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  const collateralAddress = RegTestFoundationKeys[0].owner.address
+  const futureRewardPercentage = 0.05
+  const futureInterval = 25
+
+  let tslaAddress: string
+
+  async function setup (): Promise<void> {
+    const oracleId = await testing.rpc.oracle.appointOracle(await testing.generateAddress(), [
+      {
+        currency: 'USD',
+        token: 'TSLA'
+      }
+    ],
+    { weightage: 1 }
+    )
+    await testing.generate(1)
+
+    await testing.rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), {
+      prices: [
+        {
+          currency: 'USD',
+          tokenAmount: '2@TSLA'
+        }
+      ]
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'DUSD',
+      name: 'DUSD',
+      fixedIntervalPriceId: 'DUSD/USD',
+      mintable: true,
+      interest: new BigNumber(0)
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      name: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD',
+      mintable: true,
+      interest: new BigNumber(1)
+    })
+    await testing.generate(1)
+
+    await testing.token.mint({
+      symbol: 'DUSD',
+      amount: 4
+    })
+    await testing.generate(1)
+
+    await testing.token.mint({
+      symbol: 'TSLA',
+      amount: 4
+    })
+    await testing.generate(1)
```

Comment:
need change the loan token source from "mint" to "takeLoan from vault"

Reply:
What is the reason? 

Peter does not do so in https://github.com/DeFiCh/ain/blob/master/test/functional/feature_futures.py

Maybe I miss the context as I had been taking long leave for some time

Reply:
https://github.com/DeFiCh/jellyfish/issues/865